_posts/Processing Data/2020-03-09-Exactly-Once-Example.md

```diff
@@ -32,81 +32,116 @@ from the latest Flink checkpoint in case of failures.
+If you choose to use the Gradle Build Tool, make sure the Maven repo in SDP is available to your development workstation as mentioned in the [post]({{site.baseurl}}/getting started/2020/07/14/create-flink-project-on-streaming-data-platform.html). The following is an example of ```pravega-samples/flink-connector-examples/build.gradle``` file which only shows the modified section. Since the example uses shadow JARs, make sure to add `classifier = ""` and `zip64 true` to the `shadowJar` config. 
 ```
-The app completes checkpointing at position 16, and in the meantime, continues to write to the 
-Pravega stream until it introduces an artificial exception to simulate transaction failure 
-at position 26. Upon the failure, the app restores from its last successful checkpoint 
-at position 16. Without the Pravega EXACTLY_ONCE enabled, it is likely that duplicate events, 
-from position 17 to 26, will be written to the Pravega stream. 
-
-And indeed, that's what checker app shows. Note that output of duplicate events may not necessarily 
-be within the checker start and end block. 
-
+shadowJar {
+    dependencies {
+        include dependency("org.scala-lang.modules:scala-java8-compat_${flinkScalaVersion}")
+        include dependency("io.pravega:pravega-connectors-flink-${flinkMajorMinorVersion}_${flinkScalaVersion}")
+        include dependency("io.pravega:pravega-keycloak-client:${pravegaKeycloakVersion}")
+    }
+    classifier = ""
+    zip64 true
+}
+
+publishing {
+    repositories {
+        maven {
+            url = "http://localhost:9090/maven2"
+            credentials {
+                username "desdp"
+                password "password"
+            }
+            authentication {
+                basic(BasicAuthentication)
+            }
+        }
+    }
+
+    publications {
+        shadow(MavenPublication) { publication ->
+                project.shadow.component(publication)
+        }
+    }
+}
 ```
-============== Checker starts ===============
-Duplicate event: 18
-Duplicate event: 20
-Duplicate event: 22
-Duplicate event: 24
-Duplicate event: 17
-Duplicate event: 19
-Duplicate event: 21
-Duplicate event: 23
-Found duplicates
-============== Checker ends  ===============
+The preferable **name/namespace/scope** for this project is ```exactlyonce```.  
+The **Flink Image** for creating the cluster is ```1.9.0```.  
+The **Main Class** for creating the new app is ```io.pravega.example.flink.primer.process.ExactlyOnceChecker```.   
+Please also make sure to pass the same parameters as discussed in the [original exactly-once post](https://github.com/pravega/pravega-samples/tree/master/flink-connector-examples/doc/exactly-once). The **controller** address in this case is: ```tcp://nautilus-pravega-controller.nautilus-pravega.svc.cluster.local:9090```   
+After finishing the above steps, the **State** for your Flink application should be shown as **Started**.
```

Comment:
@youminhan is the instructions to deploy flink application on SDP available somewhere else?

Reply:
maybe some screeshots?